### PR TITLE
refactor: item transformer model rework

### PR DESCRIPTION
**Pull request by P0keDev**
creates individual models for each custom ItemStack and ItemProperty, which features can selectively declare as dependencies. design isn't final and some stuff might be broken at the moment

`PowderItemStackModel` goes unused because no feature actually used that transformation - the itemstack is only for creating powders in the item guide as far as i can tell.
`IntelligenceSkillPointsItemStackModel` also goes unused, because that itemstack never had a corresponding feature. if we do go ahead with this design, i will add a feature for it in this PR

apologies for the mess of commits and whatnot, stuff got weird with the branch change

### CODE REVIEW

File: common/src/main/java/com/wynntils/wynn/model/item/properties/DungeonKeyPropertyModel.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+package com.wynntils.wynn.model.item.properties;
+
+import com.wynntils.core.managers.Model;
+import com.wynntils.wynn.item.parsers.WynnItemMatchers;
+import com.wynntils.wynn.item.properties.DungeonKeyProperty;
+import com.wynntils.wynn.model.item.ItemStackTransformManager;
+import com.wynntils.wynn.model.item.ItemStackTransformManager.ItemPropertyWriter;
+
+public class DungeonKeyPropertyModel extends Model {
+    private static final ItemPropertyWriter D_UNGEON_KEY_WRITER =
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/wynn/model/item/properties/DungeonKeyPropertyModel.java]: This got misstyped. I think.

**Comment by Incompleteusern** [comment]
This should fix #529

**Comment by kristofbolyai** [comment]
This is certainly a PR that will require discussion in Discord. But overall, I am still a fan of the idea.

**Review by magicus — COMMENTED** [review]
As I've said, I'm not convinced the idea we have of being able to turn off models is the best, but as long as we have it, this PR fits right in with that design think. So I think it is good, for now at least.

**Comment by kristofbolyai** [comment]
Hey @P0keDev! I fixed up your whole commit history, removed every other old commit. The only sad thing is that Github thinks I claimed "ownership" of your commits. Kinda. 
image
I do not know how this turns out in code statistics. Anyways, if this bothers you, I can probably fix it. 

I also seem to fixed a lot of merge issues coming from magicus' refactors of models and web profiles, which seemed to affect this PR quite a bit. It looks great now, I think.

**Comment by P0keDev** [comment]
> Hey @P0keDev! I fixed up your whole commit history, removed every other old commit. The only sad thing is that Github thinks I claimed "ownership" of your commits. Kinda. 
> image
> I do not know how this turns out in code statistics. Anyways, if this bothers you, I can probably fix it. 
> 
> I also seem to fixed a lot of merge issues coming from magicus' refactors of models and web profiles, which seemed to affect this PR quite a bit. It looks great now, I think.

thanks for cleaning it up! i don't mind the commit ownership thing so don't worry about changing it

i'll try to finalize this pr and get it merge-ready today or tomorrow since it seems everyone is okay with the direction of this design

**Comment by P0keDev** [comment]
PR should be merge-ready now, but I haven't had the opportunity to test it since I'm doing this from an airport at the moment. Feel free to review the code itself; I won't merge until I make sure everything works properly in-game

**Review by magicus — DISMISSED** [review]

**Review by magicus — DISMISSED** [review]

**Comment by kristofbolyai** [comment]
(do not merge this until further notice from P0ke, this is untested)

**Comment by P0keDev** [comment]
finally got around to testing this PR - i added the corresponding model as a dependency to the search overlay feature and registered the transform manager because i missed that somehow lol

i also removed the intelligence mana table feature because i realized it wasn't updated for 2.0. i kept the model and custom itemstack code intact for whenever that gets fixed; the feature was really just a boilerplate for the model dependency anyway. as a result of that, the intelligence points item stack matcher isn't actually used now, but i did update it to check for the right item (golden axe instead of book)

**Review by magicus — APPROVED** [review]

**Pull request merged**